pkg/user/handler.go

```diff
@@ -135,6 +137,69 @@ func (h Handler) ValidateEmail(c *gin.Context) {
 	c.Status(http.StatusOK)
 }
 
+type requestPasswordResetRequest struct {
+	Email string `json:"email" binding:"required,email"`
+}
+
+func (h Handler) RequestPasswordReset(c *gin.Context) {
+	// swagger:route POST /users/request-reset requestPasswordReset
+	//
+	// Request password reset
+	//
+	// Request user's password reset
+	//
+	// responses:
+	//   201:
+	//   400: Error
+	//   404: Error
+	//   415: Error
+	var request requestPasswordResetRequest
+	if err := handler.DataBinder(c, &request); err != nil {
+		_ = c.Error(err)
+		return
+	}
+
+	err := h.userService.RequestPasswordReset(request.Email)
+	if err != nil {
+		_ = c.Error(err)
+		return
+	}
+
+	c.Status(http.StatusCreated)
+}
+
+type resetPasswordRequest struct {
```

Comment:
Are you using this struct client side? When you construct the request, do you ensure this type is what's sent as the request body?